tests/test_fields.py

```diff
@@ -144,6 +172,149 @@ def test_CategoricalField_binary_op(self, op):
+
+        with self.subTest("Testing with numeric operation: first is CategorialField, second is CategoricalMemField"):
+            first, second = first_field, second_cat_mem_field
+            output = op(first, second)
+            self.assertIsInstance(output, fields.NumericMemField)
+            np.testing.assert_array_equal(output, expected_result)
+            self.assertEqual(output.data.dtype, expected_datatype)
+
+        with self.subTest("Testing with numeric operation to be commutative: first is CategoricalMemField, second is CategorialField"):
+            first, second = second_cat_mem_field, first_field
+            output = op(first, second)
+            self.assertIsInstance(output, fields.NumericMemField)
+            np.testing.assert_array_equal(output, expected_result)
+            self.assertEqual(output.data.dtype, expected_datatype)
+
+        with self.subTest("Testing with numeric operation: first is CategorialField, second also is CategorialField"):
+            first, second = first_field, second_field
+            output = op(first, second)
+            self.assertIsInstance(output, fields.NumericMemField)
+            np.testing.assert_array_equal(output, expected_result)
+            self.assertEqual(output.data.dtype, expected_datatype)
+            
+        with self.subTest("Testing with numeric operationto be commutative: first is CategorialField, second also is CategorialField"):
+            first, second = second_field, first_field
+            output = op(first, second)
+            self.assertIsInstance(output, fields.NumericMemField)
+            np.testing.assert_array_equal(output, expected_result)
+            self.assertEqual(output.data.dtype, expected_datatype)
+
+
+    @parameterized.expand(CATEGORY_NUMERIC_OPERATION_TEST)
+    def test_CategoricalMemField_numeric_binary_op(self, op, first_datatype, first_data, second_datatype, second_data, expected_datatype):
+        """
+        Categorical field ops against numpy, categorical memory field, categorical field
+        """
+        first_cat_mem_field = fields.CategoricalMemField(self.s, second_datatype, {"a": 1, "b": 2, "c": 3})
+        first_ndarray = np.array(first_data, dtype=first_datatype)
+        first_cat_mem_field.data.write(first_ndarray)
+
+        second_field = self.df.create_categorical('catf_2', second_datatype, {"x": 0, "y": 1, "z": 6})
+        second_cat_mem_field = fields.CategoricalMemField(self.s, second_datatype, {"x": 0, "y": 1, "z": 6})
+        second_numeric_field = self.df.create_numeric('num', second_datatype)
+        second_ndarray = np.array(second_data, dtype=second_datatype)
+        second_field.data.write(second_ndarray)
+        second_cat_mem_field.data.write(second_ndarray)
+        second_numeric_field.data.write(second_ndarray)
+
+        expected_result = op(first_ndarray, second_ndarray)
+
+        with self.subTest("Testing numeric operation: first is CategorialMemField, second is ndarray"):
```

Comment:
We could reduce all these subtest blocks down to one by iterating over the combinations of first and second that you want to test with:

```suggestion
   combinations = [
       (first_cat_mem_field, second_ndarray),
       ...
   ]
   for first, second in combinations:
        with self.subTest(f"Testing numeric operation: first is {type(first)}, second is {type(second)}"):
            output = op(first, second)
            np.testing.assert_array_equal(output, expected_result)
            self.assertEqual(output.data.dtype, expected_datatype)
```